### PR TITLE
Add CircleCI.

### PR DESCRIPTION
**Pull request by stephenplusplus**
RE: #1

@callmehiphop I think only you can enable Circle, though: https://circleci.com/add-projects/gh/callmehiphop

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,10 +4,9 @@
   "description": "Generic Pooling Library",
   "main": "index.js",
   "scripts": {
-    "cover": "nyc npm run test",
     "lint": "eslint index.js lib/*.js",
     "prettier": "prettier --write *.js lib/*.js test/*.js",
-    "test": "ava test/*.js --verbose --timeout=10s"
+    "test": "nyc ava test/*.js --verbose --timeout=10s"
```
Comment (stephenplusplus) [inline package.json]: Just calling this out. Putting it in the same command makes it easier to notice when we drop coverage.

Comment (callmehiphop) [reply package.json]: Wait, how is it easier than a separate npm script?

Comment (stephenplusplus) [reply package.json]: When we run `npm test`, we'll see the output and not have to remember to check coverage separately.

Comment (stephenplusplus) [reply package.json]: Whatever you want to do, though. `codecov` bot, coveralls, etc

Comment (callmehiphop) [reply package.json]: Fair enough! I'm into codecov, want to add the npm script to this PR and change the circle script to run coverage reports??

Comment (stephenplusplus) [reply package.json]: Still figuring this out, but I think `npm test` will always spit out the nyc report, and codecov will report on it from the newly-defined task in config.yml

**Pull request merged**